modules/ROOT/pages/functions.adoc

```diff
@@ -0,0 +1,142 @@
+  --tenant <tenant-name> \
+  --namespace default \
+  --name exclamation \
+  --inputs persistent:///default/
+  --output persistent:///default/
+----
++
++
+If the function is set up and ready to accept messages, you should see "Created Successfully!"
+
+. Use `./pulsar-admin functions list --tenant <tenant-name>` to list the functions in your tenant and confirm your new function was created.  
+
+=== Trigger with CLI
+
+To test a function with the Pulsar CLI instead of the Admin Console, send a test value with Pulsar CLI's `trigger`.
+
+. Listen for messages on the output topic: 
++
+[source, bash]
+----
+$ bin/pulsar-client consume persistent://<tenant-name>/default/<topic-name> \
+  --subscription-name my-subscription
+  --num-messages 0 # Listen indefinitely
+----
++
+. Test your exclamation function with `trigger`:
++
+[source, bash]
+----
+$ ./pulsar-admin functions trigger \
+  --name exclamation \
+  --tenant <tenant-name> \
+  --namespace default \
+  --trigger-value "Hello world"
+----
++
+The trigger sends the string `Hello world` to your exclamation function. Your function should output `Hello world!` to your consumed output. 
+
+== Add Functions using Pulsar Admin Console
+
+Add functions in the *Functions* tab of the Admin Console. 
+
+. Select *Choose File* to choose a local Function. In this example, we chose `exclamation_function.py`. Choose the file you want to pull the function from and which function you want to use within that file.
++
+image::admin-console-add-function.png[Add Function in Admin Console]
++
+To specify the class name, it depends on the programming language (or runtime) of the function. Python, Java, and Go functions are added differently from each other.
+
+Python functions are added by loading a Python file (.py) or a zipped Python file (.zip). When adding Python files, the Class Name is specified as the name of the Python file without the extension plus the class you want to execute. 
+
+For example, if the Python file is called `function.py` and the class is `exclamation`, then the class name would be `function.exclamation`. The file can contain multiple classes, but only one will be used. If there is no class in the Python file (when using a basic function, for example) just specify the filename without the extension (ex. `function`).
+
+Java functions are added by loading a Java jar file (.jar). When adding Java files, you also need to specify the name of the class to execute as the function. 
+
+Go functions are added by loading a Go file (.go). For more information on packaging a Go function, see https://pulsar.apache.org/docs/en/functions-package/#go[Packaging Functions in Go^].
+
+[start=3]
+. Choose your function's name and namespace. 
+. Your input topics, output topics, log topics, and processing guarantees will auto-populate. 
+
+. Provide a *Configuration Key* in the dropdown menu. For a list of configuration keys, see the https://pulsar.apache.org/functions-rest-api/#[Pulsar Docs].
```

Comment:
Are we talking about this Pulsar doc page?
https://pulsar.apache.org/functions-rest-api/#operation/registerFunction